heron/statemgrs/src/python/configloader.py

```diff
@@ -0,0 +1,89 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#!/usr/bin/env python2.7
+''' configloader.py '''
+
+import os
+import re
+import sys
+import yaml
+
+def load_state_manager_locations(cluster, state_manager_config_file='heron-conf/statemgr.yaml'):
+  """ Reads configs to determine which state manager to use and converts them to state manager
+  locations. Handles a subset of config wildcard substitution supported in the substitute method in
+  com.twitter.heron.spi.common.Misc.java"""
+  with open(state_manager_config_file, 'r') as stream:
+    config = yaml.load(stream)
+
+  home_dir = os.getenv('HOME')
+  wildcards = {
+      "~" : home_dir,
+      "${HOME}" : home_dir,
+      "${CLUSTER}" : cluster,
+  }
+  if os.getenv('JAVA_HOME'):
+    wildcards["${JAVA_HOME}"] = os.getenv('JAVA_HOME')
+
+  config = __replace(config, wildcards, state_manager_config_file)
+
+  # need to convert from the format in statemgr.yaml to the format that the python state managers
+  # takes. first, set reasonable defaults to local
+  state_manager_location = {
+      'type': 'file',
+      'name': 'local',
+      'tunnelhost': 'localhost',
+      'rootpath': '~/.herondata/repository/state/local',
+  }
+
+  # then map the statemgr.yaml config keys to the python state manager location
+  key_mappings = {
+      'heron.statemgr.connection.string': 'hostport',
+      'heron.statemgr.tunnel.host': 'tunnelhost',
+      'heron.statemgr.root.path': 'rootpath',
+  }
+  for config_key in key_mappings:
+    if config_key in config:
+      state_manager_location[key_mappings[config_key]] = config[config_key]
+
+  state_manager_class = config['heron.class.state.manager']
+  if state_manager_class == 'com.twitter.heron.statemgr.zookeeper.curator.CuratorStateManager':
+    state_manager_location['type'] = 'zookeeper'
+    state_manager_location['name'] = 'zk'
+
+  return [state_manager_location]
+
+def __replace(config, wildcards, config_file):
+  """For each kvp in config, do wildcard substitution on the values"""
+  for config_key in config:
+    config_value = config[config_key]
+    original_value = config_value
+    if isinstance(config_value, basestring):
+      for token in wildcards:
+        print("config_value: %s, token: %s wildcard value: %s" %
+              (config_value, token, wildcards[token]))
+        if wildcards[token]:
+          config_value = config_value.replace(token, wildcards[token])
+      found = re.findall(r'\${[A-Z_]+}', config_value)
+      if found:
+        raise ValueError("%s=%s in file %s contains unsupported or unset wildcard tokens: %s" %
+                         (config_key, original_value, config_file, ", ".join(found)))
+      config[config_key] = config_value
+  return config
+
+if __name__ == "__main__":
+  if len(sys.argv) > 1:
+    locations = load_state_manager_locations('local', sys.argv[1])
```

Comment:
Yes, this was just for an easy way to manually check a file besides those in the unit tests.